sysdig/getmetrics_sysdig.py

```diff
@@ -0,0 +1,348 @@
+        try:
+            sysdig_api_key = config_parser.get('sysdig', 'api_key')
+            hostname = config_parser.get('sysdig', 'hostname')
+            all_metrics = config_parser.get('sysdig', 'all_metrics').split(',')
+            sysdig_http_proxy = config_parser.get('sysdig', 'sysdig_http_proxy')
+            sysdig_https_proxy = config_parser.get('sysdig', 'sysdig_https_proxy')
+            sysdig_metric_chunk_size= config_parser.get('sysdig', 'metric_chunk_size')
+            sysdig_host_chunk_size=config_parser.get('sysdig', 'host_chunk_size')
+        except ConfigParser.NoOptionError:
+            logger.error(
+                "Agent not correctly configured. Check config file.")
+            sys.exit(1)
+
+        if len(sysdig_api_key) == 0:
+            logger.warning(
+                "Agent not correctly configured(API KEY). Check config file.")
+            exit()
+        if len(hostname) == 0:
+            logger.warning(
+                "Agent not correctly configured. Check config file.")
+            exit()
+
+        sysdig_config = {
+            "sysdig_api_key": sysdig_api_key,
+            "hostname": hostname,
+            "all_metrics": all_metrics,
+            "httpProxy": sysdig_http_proxy,
+            "httpsProxy": sysdig_https_proxy,
+            "host_chunk_size":sysdig_host_chunk_size,
+            "metric_chunk_size":sysdig_metric_chunk_size
+        }
+    else:
+        logger.warning("No config file found. Exiting...")
+        exit()
+
+    return sysdig_config
+
+
+def format_data(res,sub_metric_list):
+    formated_data = []
+
+    for data_dict in res['data']:
+
+        instance = data_dict['d'][0] + '_' + data_dict['d'][1]
+
+        metric_data_dict={}
+
+        if sub_metric_list== ['container.id','host.hostName']:
+            iterator =0
+        else:
+            iterator =2
+
+        for metric in sub_metric_list:
+            # if(metric not in ['container.id','host.hostName']):
+                metric_data_dict[metric + '[' + instance + ']'] = str(data_dict['d'][iterator])
+                iterator += 1
+
+        metric_data_dict['timestamp']=str(data_dict['t']*1000)
+        formated_data.append(metric_data_dict)
+
+    return formated_data
+
+
+def send_data(chunk_metric_data):
+    send_data_time = time.time()
+    # prepare data for metric streaming agent
+    to_send_data_dict = dict()
+    to_send_data_dict["metricData"] = json.dumps(chunk_metric_data)
+    to_send_data_dict["licenseKey"] = agent_config_vars['licenseKey']
+    to_send_data_dict["projectName"] = agent_config_vars['projectName']
+    to_send_data_dict["userName"] = agent_config_vars['userName']
+    to_send_data_dict["instanceName"] = socket.gethostname().partition(".")[0]
+    to_send_data_dict["samplingInterval"] = agent_config_vars['samplingInterval']
+    to_send_data_dict["agentType"] = "custom"
+
+    to_send_data_json = json.dumps(to_send_data_dict)
```

Comment:
why do json.dumps here when you do json.loads before sending?

Reply:
corrected.